### PR TITLE
Recovering TTree baskets

### PR DESCRIPTION
**Pull request by tamasgal**
This PR implements the recovery mechanics of TBasket data.

### CODE REVIEW

File: src/bootstrap.jl [hunk 1]
```diff
@@ -2,7 +2,8 @@
 # dynamically in future.
 
 struct RecoveredTBasket
-    data
+    data::Vector{UInt8}
+    offsets::Vector{UInt32}
```
Comment (Moelf) [inline src/bootstrap.jl:6]: this looks like the underlying data for a 1-jagged array?

**Comment by Moelf** [comment]
interesting, so to put this mechanism in words, the recovery basket is the "last basket" in branch (makes sense, it's was being written when crash happened), which means:

1. we can't find it in `branch.fEntry`:
https://github.com/JuliaHEP/UnROOT.jl/blob/70732992b3ca6c47732ce9cea1ce1ef341672fe5/src/iteration.jl#L189-L191
2. but we will find it at the end of `branch.fBasket`:
https://github.com/JuliaHEP/UnROOT.jl/blob/70732992b3ca6c47732ce9cea1ce1ef341672fe5/src/iteration.jl#L56-L60

**Comment by tamasgal** [comment]
So currently it's usually empty, but when there are recovered baskets, it's filled with `missing` (with the same number of of `missing` as there are "real" baskets) and at the end is a `RecoveredBasket`.

File: src/bootstrap.jl (hunk 1 above)
Comment (tamasgal) [reply src/bootstrap.jl:6]: that can be jagged or not, similar to the usual basket. If `offsets` is non-empty, it's jagged

**Comment by Moelf** [comment]
ok! so concretely there should only be at most 1 real `fBaskets.elements` that's useful, this makes sense from ROOT writer perspective

**Pull request merged**